del_section/src/comp.php

```diff
@@ -0,0 +1,278 @@
+
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL);
+
+echo "<p>Starting</p>";
+
+//*********GET DATE**********************
+
+$date = date('Y-m-d\TH:i:s');
+// $date1 = "2015-12-21T15:44:36";
+
+
+//*****************GRAB_INPUT_DATA**********
+
+$uploaddir = '../uploads/';
+$uploadfile = $uploaddir . basename($_FILES['userfile']['name']);
+
+echo '<pre>';
+if (move_uploaded_file($_FILES['userfile']['tmp_name'], $uploadfile)) {
+	echo "File is valid, and was successfully uploaded.\n";
+} else {
+	echo "File is invalid, and failed to upload - Please try again. -\n";
+}
+echo "</br>";
+print_r($uploadfile);
+echo "</br>";
+echo "</br>";
+
+/**
+ * Get the user config file. This script will fail disgracefully if it has not been created and nothing will happen.
+**/
+
+require('../../user.config.php');
+
+echo "Tenancy Shortcode set: " . $shortCode;
+echo "</br>";
+
+echo "Client ID set: " . $clientID;
+echo "</br>";
+
+echo "User GUID to use: " . $TalisGUID;
+echo "</br>";
+
+$shouldPublishLists = filter_var($_REQUEST['PUBLISH_LISTS'], FILTER_VALIDATE_BOOLEAN) || FALSE;
+
+echo "Should publish lists?: " . var_export($shouldPublishLists, true);
+echo "</br>";
+
+if(isset($_REQUEST['DRY_RUN']) &&
+	$_REQUEST['DRY_RUN'] == "writeToLive") {
+		$shouldWritetoLive = "true";
+	}
+	else
+	{
+		$shouldWritetoLive = "false";
+	}
+
+echo "Writing to live tenancy?: $shouldWritetoLive";
+echo "</br>";
+echo "</br>";
+
+//**********CREATE LOG FILE TO WRITE OUTPUT*
+
+$myfile = fopen("../../report_files/delsection_output.log", "a") or die("Unable to open delsection_output.log");
+fwrite($myfile, "Started | Input File: $uploadfile | Date: " . date('d-m-Y H:i:s') . "\r\n\r\n");
+fwrite($myfile, "List name" . "\t" . "List ID" . "\t" . "Section GUID" . "\t" . "Section deleted" . "\t" . "List Published" . "\r\n");
+
+$tokenURL = 'https://users.talis.com/oauth/tokens';
+$content = "grant_type=client_credentials";
+
+//************GET_TOKEN***************
+
+
+$ch = curl_init();
+
+curl_setopt($ch, CURLOPT_URL, $tokenURL);
+curl_setopt($ch, CURLOPT_POST, TRUE);
+curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+curl_setopt($ch, CURLOPT_USERPWD, "$clientID:$secret");
+curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
+
+$return = curl_exec($ch);
+$info = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+if ($info !== 200){
+	echo "<p>    ERROR: There was an error getting a token:</p><pre>" . var_export($return, true) . "</pre>";
+} else {
+	echo "    Got Token</br>";
+}
+
+curl_close($ch);
+
+$jsontoken = json_decode($return);
+
+if (!empty($jsontoken->access_token)){
+	$token = $jsontoken->access_token;
+} else {
+	echo "<p>    ERROR: Unable to get an access token</p>";
+	exit;
+}
+
+
+//***********READ**DATA******************
+
+$file_handle = fopen($uploadfile, "rb");
+
+while (!feof($file_handle) )  {
+
+	$line_of_text = fgets($file_handle);
+	$parts = explode(" ", $line_of_text);
+	$barc = trim($parts[0]);
```

Comment:
Not sure what `$barc` is? If it is the guid of the section, can we call it `section_guid`? 
This is seen a number of places throughout the script.

Reply:
mm, $barc originally stood for barcode.. taken from an earlier php tool I wrote for our lms a long time ago.. 

It's just the variable name for the uuid pulled out of the input file. I agree this should be changed to something more relevant! section_guid is ideal :))